bitcoin/src/crypto/key.rs

```diff
@@ -331,14 +318,6 @@ impl fmt::Display for CompressedPublicKey {
     }
 }
 
-impl FromStr for CompressedPublicKey {
-    type Err = Error;
-
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
-        CompressedPublicKey::from_slice(&<[u8; 33]>::from_hex(s)?).map_err(Into::into)
-    }
-}
-
 impl TryFrom<PublicKey> for CompressedPublicKey {
     type Error = UncompressedPubkeyError;
```

Comment:
Perhaps this should be called `UncompressedPublicKeyError` to be uniform with `ParsePublicKeyError` and `ParseCompressedPublicKeyError`.